src/intrinsic/archs.rs

```diff
@@ -2935,31 +1508,1475 @@ match name {
+    "llvm.x86.xop.vprotw" => "__builtin_ia32_vprotw",
+    "llvm.x86.xop.vprotwi" => "__builtin_ia32_vprotwi",
+    "llvm.x86.xop.vpshab" => "__builtin_ia32_vpshab",
+    "llvm.x86.xop.vpshad" => "__builtin_ia32_vpshad",
+    "llvm.x86.xop.vpshaq" => "__builtin_ia32_vpshaq",
+    "llvm.x86.xop.vpshaw" => "__builtin_ia32_vpshaw",
+    "llvm.x86.xop.vpshlb" => "__builtin_ia32_vpshlb",
+    "llvm.x86.xop.vpshld" => "__builtin_ia32_vpshld",
+    "llvm.x86.xop.vpshlq" => "__builtin_ia32_vpshlq",
+    "llvm.x86.xop.vpshlw" => "__builtin_ia32_vpshlw",
+    "llvm.x86.xtest" => "__builtin_ia32_xtest",
     // xcore
     "llvm.xcore.bitrev" => "__builtin_bitrev",
     "llvm.xcore.getid" => "__builtin_getid",
     "llvm.xcore.getps" => "__builtin_getps",
     "llvm.xcore.setps" => "__builtin_setps",
-    // ptx
-    "llvm.ptx.bar.sync" => "__builtin_ptx_bar_sync",
-    "llvm.ptx.read.clock" => "__builtin_ptx_read_clock",
-    "llvm.ptx.read.clock64" => "__builtin_ptx_read_clock64",
-    "llvm.ptx.read.gridid" => "__builtin_ptx_read_gridid",
-    "llvm.ptx.read.laneid" => "__builtin_ptx_read_laneid",
-    "llvm.ptx.read.lanemask.eq" => "__builtin_ptx_read_lanemask_eq",
-    "llvm.ptx.read.lanemask.ge" => "__builtin_ptx_read_lanemask_ge",
-    "llvm.ptx.read.lanemask.gt" => "__builtin_ptx_read_lanemask_gt",
-    "llvm.ptx.read.lanemask.le" => "__builtin_ptx_read_lanemask_le",
-    "llvm.ptx.read.lanemask.lt" => "__builtin_ptx_read_lanemask_lt",
-    "llvm.ptx.read.nsmid" => "__builtin_ptx_read_nsmid",
-    "llvm.ptx.read.nwarpid" => "__builtin_ptx_read_nwarpid",
-    "llvm.ptx.read.pm0" => "__builtin_ptx_read_pm0",
-    "llvm.ptx.read.pm1" => "__builtin_ptx_read_pm1",
-    "llvm.ptx.read.pm2" => "__builtin_ptx_read_pm2",
-    "llvm.ptx.read.pm3" => "__builtin_ptx_read_pm3",
-    "llvm.ptx.read.smid" => "__builtin_ptx_read_smid",
-    "llvm.ptx.read.warpid" => "__builtin_ptx_read_warpid",
-    // cuda
-    "llvm.cuda.syncthreads" => "__syncthreads",
-_ => unimplemented!("***** unsupported LLVM intrinsic {}", name),
+    // Intrinsics not bound to a specific arch
+    "llvm.adjust.trampoline" => "__builtin_adjust_trampoline",
+    "llvm.debugtrap" => "__builtin_debugtrap",
+    "llvm.eh.unwind.init" => "__builtin_unwind_init",
+    "llvm.flt.rounds" => "__builtin_flt_rounds",
+    "llvm.init.trampoline" => "__builtin_init_trampoline",
+    "llvm.objectsize.i16.p0i8" => "__builtin_object_size",
+    "llvm.objectsize.i32.p0i8" => "__builtin_object_size",
+    "llvm.objectsize.i64.p0i8" => "__builtin_object_size",
+    "llvm.objectsize.i8.p0i8" => "__builtin_object_size",
+    "llvm.objectsize.v16i8.p0i8" => "__builtin_object_size",
+    "llvm.objectsize.v2i64.p0i8" => "__builtin_object_size",
+    "llvm.objectsize.v4i32.p0i8" => "__builtin_object_size",
+    "llvm.objectsize.v8i16.p0i8" => "__builtin_object_size",
+    "llvm.stackrestore" => "__builtin_stack_restore",
+    "llvm.stacksave" => "__builtin_stack_save",
+    "llvm.trap" => "__builtin_trap",
```

Comment:
I don't think any of these are used by stdarch or any other part of the standard library.

Reply:
It is only used inside cg_llvm as implementation for the `abort` builder function. cg_gcc directly calls the `abort` builtin function there.

Reply:
`llvm.debugtrap` is needed by the `breakpoint` intrinsic.

Reply:
Wouldn't the implementation of said intrinsic directly call the right gcc builtin?